pkg/runtime/framework/plugins/jobset/jobset.go

```diff
@@ -153,6 +156,42 @@ func (j *JobSet) Validate(info *runtime.Info, _, newObj *trainer.TrainJob) (admi
 	return nil, allErrs
 }
 
+func (j *JobSet) checkPodSpecOverridesImmutability(ctx context.Context, oldObj, newObj *trainer.TrainJob) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if oldObj == nil {
+		// Checking immutability makes only sense on updates
+		return allErrs
+	}
+
+	jobSet := &jobsetv1alpha2.JobSet{}
+	if err := j.client.Get(ctx, client.ObjectKeyFromObject(newObj), jobSet); err != nil && !apierrors.IsNotFound(err) {
+		allErrs = append(allErrs, field.InternalError(podSpecOverridePath, err))
+	} else {
+		changed := !equality.Semantic.DeepEqual(oldObj.Spec.PodSpecOverrides, newObj.Spec.PodSpecOverrides)
+		suspended := newObj.Spec.Suspend != nil && *newObj.Spec.Suspend
+		if changed {
+			if !suspended {
+				allErrs = append(allErrs, field.Forbidden(podSpecOverridePath, "PodSpecOverrides can only be modified when the TrainJob is suspended"))
+			} else if !apierrors.IsNotFound(err) {
```

Comment:
```suggestion
			} else {
```

Why do we need to check if error is not found error? IIUC, we already performed the same verification in Line168.